File: src/problems/leetcode-cn/120. 三角形最小路径和/solution/java/Solution.java
class Solution {
    public int minimumTotal(List<List<Integer>> triangle) {
        for (int i = triangle.size() - 2; i >= 0; i--) {
            List<Integer> row = triangle.get(i);
            for (int j = 0; j < row.size(); j++) {
                int n1 = triangle.get(i+1).get(j), n2 = triangle.get(i+1).get(j+1);
                row.set(j, row.get(j) + (n1 <= n2 ? n1 : n2));
            }
            triangle.set(i, row);
        }
        return triangle.get(0).get(0);
    }
}

File: src/problems/leetcode-cn/120. 三角形最小路径和/solution/go/solution.go
func minimumTotal(triangle [][]int) int {
    rows := len(triangle)
    for r := rows - 2; r >= 0; r-- {
        cols := len(triangle[r])
        for c := 0; c < cols; c++ {
            triangle[r][c] += min(triangle[r+1][c], triangle[r+1][c+1])
        }
    }
    return triangle[0][0]
}

func min(a, b int) int {
    if a <= b {
        return a
    }
    return b
}